samples/WebApi/README.md

```diff
@@ -0,0 +1,104 @@
+
+## Create your app 
+
+Open a **new** shell/command prompt window and run the following commands. If using Windows make sure to use ``x64 Native Tools Command Prompt for VS 2017`` .
+
+``> dotnet new webapi -o myApp`` 
+
+``> cd myApp``
+
+## Add CoreRT to your project
+
+Using CoreRT to compile your application is done via the ILCompiler NuGet package.
+
+In your shell/command prompt navigate to the root directory of your project run the command 
+
+`` dotnet new nuget ``
+
+This will add a nuget.config file to your application. Open the file and in ``<packageSources> `` under ``<clear/>`` add the following:
+
+``<add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" /> ``
+
+Once you've added the package source, add a reference to the compiler by running the following command:
+
+`` dotnet add package Microsoft.DotNet.ILCompiler -v <Version> `` 
+
+where ``<Version>`` is the [latest version of the ILCompiler package](https://dotnet.myget.org/feed/dotnet-core/package/nuget/Microsoft.DotNet.ILCompiler) (e.g. 1.0.0-alpha-25920-0).
+
+After the package has been succesfully added to your project, open the file called ``Startup.cs`` and in the ``ConfigureServices`` method modify the line:    
+    
+    services.AddMvc();
+
+to
+
+    services.AddMvcCore().AddJsonFormatters();
+
+## Using reflection 
+Runtime directives are XML configuration files, which specify which elements of your program are available for reflection. They are used at compile-time to enable AOT compilation in applications at runtime. 
+
+If your application makes use of reflection, you will need to create a rd.xml file specifying explicitly, which assemblies and types should be made available. For example, in  your .NET Core Web API application, reflection is required to determine the correct namespace, from which to load the ``Startup`` type. Both are defined respectively via the ``<Assembly> `` and ``<Type>`` attributes. For example, in the case of our specific application;
+
+    <Assembly Name="SampleWebApi">
+      <Type Name="SampleWebApi.Startup" Dynamic="Required All" />
+    </Assembly>
+
+At runtime, if a method or type is not found or cannot be loaded, an exception will be thrown. The exception message will contain information on the missing type reference, which you can then add to the rd.xml of your program.
+In the sample app a basic rd.xml file has been added for the sample Web API application under the `build` folder. 
+
+Once you've created a rd.xml file, navigate to the root directory of your project and open its ``.csproj `` file and in the first ``<PropertyGroup>`` add the following:
+
+    <RdXmlFile>path_to_rdxml_file\rd.xml</RdXmlFile>
+
+where path_to_rdxml_file is the location of the file on your disk. Under the second ``<ItemGroup>`` remove the reference to ``Microsoft.AspNetCore.All`` and substitute it with:
+
+    <PackageReference Include="Microsoft.AspNetCore" Version="2.0.0-preview1-final" /> 
```

Comment:
The version for ASP.NET packages should be Version="2.0.0". What you have is a pre-release versions.